cluster/local/integration_tests.sh

```diff
@@ -0,0 +1,278 @@
+            exit -1
+        else
+            echo_step_completed
+        fi 
+
+        echo_info "check if has restarts"
+        if (( $(echo "$pod_stat" | awk '{print $4}') > 0 )); then
+            echo_error "has restarts"
+            exit -1
+        else
+            echo_step_completed
+        fi
+        echo
+    done <<< "$(echo "$pods" | awk 'FNR>1')"
+}
+
+# ------------------------------
+projectdir="$( cd "$( dirname "${BASH_SOURCE[0]}")"/../.. && pwd )"
+
+# get the build environment variables from the special build.vars target in the main makefile
+eval $(make --no-print-directory -C ${projectdir} build.vars)
+
+# ------------------------------
+
+HELM_VERSION="helm-v2.10.0"
+HOST_PLATFORM="${HOSTOS}_${HOSTARCH}"
+TOOLS_DIR="${CACHE_DIR}/tools"
+TOOLS_HOST_DIR="${TOOLS_DIR}/${HOST_PLATFORM}"
+HELM="${TOOLS_HOST_DIR}/${HELM_VERSION}"
+echo_step "installing helm ${HOSTOS}-${HOSTARCH}"
+mkdir -p ${TOOLS_HOST_DIR}/tmp-helm
+curl -fsSL https://storage.googleapis.com/kubernetes-helm/${HELM_VERSION}-${HOSTOS}-${HOSTARCH}.tar.gz | tar -xz -C ${TOOLS_HOST_DIR}/tmp-helm
+mv ${TOOLS_HOST_DIR}/tmp-helm/${HOSTOS}-${HOSTARCH}/helm ${HELM}
+rm -fr ${TOOLS_HOST_DIR}/tmp-helm
+echo_success "installing helm ${HOSTOS}-${HOSTARCH}"
+
+HOSTARCH="${HOSTARCH:-amd64}"
+BUILD_IMAGE="${BUILD_REGISTRY}/${PROJECT_NAME}-${HOSTARCH}"
+
+version_tag="$(cat ${projectdir}/_output/version)"
+# tag as master so that config/stack/install.yaml can be used
+STACK_IMAGE="${DOCKER_REGISTRY}/${PROJECT_NAME}:master"
+K8S_CLUSTER="${K8S_CLUSTER:-${BUILD_REGISTRY}-INTTESTS}"
+
+CROSSPLANE_NAMESPACE="crossplane-system"
+STACK_NAME="stack-azure"
+STACK_NAMESPACE="azure"
+
+# cleanup on exit
+if [ "$skipcleanup" != true ]; then
+  function cleanup {
+    echo_step "Cleaning up..."
+    export KUBECONFIG=
+    "${KIND}" delete cluster --name="${K8S_CLUSTER}"
+  }
+
+  trap cleanup EXIT
+fi
+
+echo_step "creating k8s cluster using kind"
+"${KIND}" create cluster --name="${K8S_CLUSTER}"
+export KUBECONFIG="$("${KIND}" get kubeconfig-path --name="${K8S_CLUSTER}")"
+
+# tag stack image and load it to kind cluster
+docker tag "${BUILD_IMAGE}" "${STACK_IMAGE}"
+"${KIND}" load docker-image "${STACK_IMAGE}" --name="${K8S_CLUSTER}"
+
+echo_step "installing tiller"
+"${KUBECTL}" apply -f "${projectdir}/cluster/local/helm-rbac.yaml"
+"${HELM}" init --service-account tiller
+# waiting for deployment "tiller-deploy" rollout to finish
+"${KUBECTL}" -n kube-system rollout status deploy/tiller-deploy --timeout=2m
+
+# wait for kind pods
+echo_step "wait for kind pods"
+kindpods=("kube-apiserver-${BUILD_REGISTRY}-inttests-control-plane" "kube-controller-manager-${BUILD_REGISTRY}-inttests-control-plane" "kube-scheduler-${BUILD_REGISTRY}-inttests-control-plane")
+wait_for_pods_in_namespace 120 "kube-system" "${kindpods[@]}"
+
+# install crossplane from alpha channel
+echo_step "installing crossplane from alpha channel"
+"${HELM}" repo add crossplane-alpha https://charts.crossplane.io/alpha
```

Comment:
If we are testing from the alpha channel, how do we test a version which we are working on locally? Or a version on a feature branch which is in review?

Reply:
Right now the integration tests only run against alpha, but I think enabling other revisions would be an enhancement. These integration tests are written to do the bare minimum of making sure that install and uninstall are successful. If there is desire to test against other local crossplane branches then this portion could be replaced for local e2e. As with the comment above, I think that there is a need to develop a much more robust testing engine in the future for stacks.